RetailCoder.VBE/UI/Command/FindAllReferencesCommand.cs

```diff
@@ -154,15 +156,58 @@ private SearchResultsViewModel CreateViewModel(Declaration declaration)
 
         private Declaration FindTarget(object parameter)
         {
-            var declaration = parameter as Declaration;
-            if (declaration != null)
+            if (parameter is Declaration declaration)
             {
                 return declaration;
             }
 
+            return _vbe.ActiveCodePane != null && (_vbe.SelectedVBComponent?.HasDesigner ?? false)
+                    ? FindFormDesignerTarget()
+                    : FindCodePaneTarget();
+        }
+
+        private Declaration FindCodePaneTarget()
+        {
             return _state.FindSelectedDeclaration(_vbe.ActiveCodePane);
         }
 
+        private Declaration FindFormDesignerTarget(QualifiedModuleName? qualifiedModuleName = null)
+        {            
+            (var projectId, var component)
+                = qualifiedModuleName.HasValue
+                                     ? (qualifiedModuleName.Value.ProjectId, qualifiedModuleName.Value.Component)
+                                     : (_vbe.ActiveVBProject.ProjectId, _vbe.SelectedVBComponent);
+
+            if (component?.HasDesigner ?? false)
+            {
+                if (qualifiedModuleName.HasValue)
+                {
+                    return _state.DeclarationFinder
+                                 .MatchName(qualifiedModuleName.Value.Name)
+                                 .SingleOrDefault(m => m.ProjectId == projectId
+                                                    && m.DeclarationType.HasFlag(qualifiedModuleName.Value.ComponentType)
+                                                    && m.ComponentName == component.Name);
+                }
+
+                var selectedCount = component.SelectedControls.Count;                
+                if (selectedCount > 1) { return null; }
+
+                // Cannot use DeclarationType.UserForm, parser only assigns UserForms the ClassModule flag
+                (var selectedType, var selectedName)
+                    = selectedCount == 0
+                                     ? (DeclarationType.ClassModule, component.Name)
+                                     : (DeclarationType.Control, component.SelectedControls[0].Name);
```

Comment:
Again - I'd put the assignment on the same line as the tuple definition, and have the ternaries at only 1 level of indentation:

```csharp
(var selectedType, var selectedName) = selectedCount == 0
    ? (DeclarationType.ClassModule, component.Name)
    : (DeclarationType.Control, component.SelectedControls[0].Name);
```